### ISSUE TITLE
Cannot find third part package in vscode

### ISSUE THREAD

**Issue by kenonchain** [scaffold-eth/scaffold-eth#388]
~~We have a workspace named "hardhat", but this is also the name a third-part library hardhat we are depending on. 
That makes IDE confused when it comes to code references..~~

~~For example~~ in VSCode:
![image](https://user-images.githubusercontent.com/84996642/132421838-3e03bf53-ade9-4db9-a7c4-994b70bb011a.png)

**Comment by kenonchain** [comment]
Ok I find out we can config the dependencies Directory using this extension https://marketplace.visualstudio.com/items?itemName=JuanBlanco.solidity
But this is only for VSCode. Wonder if other editor/IDE users have similar issues?
![image](https://user-images.githubusercontent.com/84996642/132437752-9f3a2aea-38b4-44c1-ab90-6ce70fe3a106.png)

**Comment by codenamejason** [comment]
@kenonchain that's a good question. Can we research the other IDE's? Also we should look into the side effects of changing the name of the folder.

**Comment by kenonchain** [comment]
Hi @codenamejason , turned out I was wrong about the name conflict. We are using `@scaffold-eth/hardhat` as package name, so couldn't conflict with `hardhat`. 

And after the laster merge, I find out we have a vscode setting pushed in.
https://github.com/austintgriffith/scaffold-eth/blob/master/.vscode/settings.json#L2

Change this line to `"solidity.packageDefaultDependenciesDirectory": "node_modules/@scaffold-eth/hardhat/node_modules/",` fix my problem.

Hi @ShravanSunder , I noticed it's your PR pushed in `.vscode/settings.json` file
Does the "can't find package redline" happen to you also? Should I put a PR on this?

**Comment by ShravanSunder** [comment]
I don’t have this issue. I think I’ve set the default directory at some point.
however feel free to PR. I think your solution works.

**Comment by codenamejason** [comment]
@kenonchain are you going to submit a PR?

**Comment by kenonchain** [comment]
> @kenonchain are you going to submit a PR?

Hi @codenamejason , thanks for reminding me. PR's up.

**Comment by StErMi** [comment]
Uhm I don't think that this is an issue or at least I never found it and I can assure that I'm setupping tons of new project :D
It seems that it's more a problem during the installation/reading dependencies

**Comment by gdnathan** [comment]
Hi ! I saw the vscode/settings.json, and i was wondering if there was a similar solution for others IDE ? I have import errors on my contract that i cant fix.
I think i need this line: 

`   "solidity.packageDefaultDependenciesDirectory": "node_modules/@scaffold-eth/hardhat/node_modules/",`

but it is vscode specific. I use neovim, with solang (or solc, i can easily switch) as Language server

**Comment by dsimmons** [comment]
@gdnathan I too just ran into this using Neovim + the `solc` language server! After some trial and error (and a not-so-brief detour from level 1 :sweat_smile:), I figured out a solution that seems to be working:

```Lua
local util = require 'lspconfig.util'
lspconfig.solc.setup {
  -- on_attach = on_attach,
  root_dir = util.root_pattern({'hardhat', 'node_modules'}),
}
```

I've commented out the `on_attach` callback because mine is defined elsewhere in the file (and yours will likely differ).

Basically, you're just instructing the `solc --lsp` to set a different project root versus the default (where the `.git` folder is located). It seems to be working for me now!

Context: This assumes you're using [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig) to configure your LSP servers. It also assumes you're using `solc` (out of the supported Solidity LSPs) and have that installed properly.
